src/core/components/AMInstallButton/index.js

```diff
@@ -378,6 +405,11 @@ const AMInstallButton: React.ComponentType<Props> = compose(
   withRouter,
   connect(mapStateToProps),
   translate(),
+  withExperiment({
```

Comment:
Interesting. I am wondering if this could introduce some sort of race condition given that this HOC will be rendered twice on a detail page..

Reply:
I added some log calls in the HOC and I can clearly see it called twice. However, the value was generated only once: I was not able to trigger a race condition where the HOC would generate a value twice.

That being said, if I comment the `cookie.set` line, then the HOC sometimes generate different values for a same HTTP request. In other words, *if* the second HOC instantiation starts creating a new value before the first HOC instantiation writes the cookie (race condition), it is possible that we get two different values and the second value will be the one stored into the cookie.

I don't know if it matters much, though. I think it would bias the data if it was easy to trigger the race condition, but that does not seem possible locally.

Reply:
> Does that seem like it would a) work and b) be worth doing? Or do you think we should just leave it as is?

a) maybe
b) mmh not sure, I am not a huge fan of yet another redux state

I would solve the issue by lifting the HOC up to `Addon` and then pass the props/values to both `AMInstallButton` and `InstallWarning` if I had to fix this issue myself :)

Reply:
It would be a problem everywhere we have both the banner and the install button, but we do have a common ancestor in all cases, so we'd have to lift the HOC up in `Addon`, `GuideAddonsCard` and `AddonVersionCard` if I am not mistaken.

Reply:
> I guess the bigger question is whether we need to do anything about it at all?

That'd probably be a good idea to care about it in order to avoid any random behavior in the future.